src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/KpiService.kt

```diff
@@ -18,31 +18,33 @@ class KpiService(
   private val offenderSearchService: OffenderSearchService,
 ) {
 
-  fun getNumberOfReviewsConductedAndPrisonersReviewed(day: LocalDate): ReviewsConductedPrisonersReviewed = runBlocking {
-    kpiRepository.getNumberOfReviewsConductedAndPrisonersReviewed(day)
+  suspend fun getNumberOfReviewsConductedAndPrisonersReviewed(day: LocalDate): ReviewsConductedPrisonersReviewed {
+    return kpiRepository.getNumberOfReviewsConductedAndPrisonersReviewed(day)
   }
 
-  fun getNumberOfPrisonersOverdue(): Int = runBlocking {
+  suspend fun getNumberOfPrisonersOverdue(): Int {
     // Get a set containing all the prisonerNumber of everyone in prison (any prison)
     val allPrisoners = mutableSetOf<String>()
-    val prisonersInPrisons = getPrisons().map { prisonId -> async { getPrisonersInPrison(prisonId) } }.awaitAll()
-    prisonersInPrisons.forEach { prisoners ->
+    getPrisons().map { prisonId ->
+      getPrisonersInPrison(prisonId)
+    }.collect { prisoners ->
```

Comment:
Turning `getPrisons()` into a Flow means that the suspend function `getPrisonersInPrison` can be called on each item without launching deferred jobs using `async`

Reply:
I don’t think we need a wrapping `coroutineScope` to handle anything since a `Flow` deals with cancellation already

Reply:
Hang on, but why not launch the coroutines? Wouldn't this make the requests to get the prisoners in each prison _**sequential**_, rather than concurrently?

These requests should be independent and it would be best to do them all "at the same time" no?

Reply:
yeah you're very likely right! because Flows are cold, it's likely that `getPrisonersInPrison` would only be called on each item after the previous one is done

REVERTED

Reply:
I mean this all happens in the background and hopefully it would still be quick enough to complete within the task allowed timeframe...but it feels right to make these requests concurrently as they're all independent.